zio-http/src/main/scala/zio/http/QueryParams.scala

```diff
@@ -88,25 +89,50 @@ final case class QueryParams(map: Map[String, Chunk[String]]) {
    */
   def getAll(key: String): Option[Chunk[String]] = map.get(key)
 
+  /**
+   * Retrieves all typed query parameter values having the specified name.
+   */
+  def getAllAs[A](key: String)(implicit codec: TextCodec[A]): Option[Chunk[A]] =
+    map.get(key).map(_.flatMap(codec.decode))
+
   /**
    * Retrieves the first query parameter value having the specified name.
    */
   def get(key: String): Option[String] = getAll(key).flatMap(_.headOption)
 
+  /**
+   * Retrieves the first typed query parameter value having the specified name.
+   */
+  def getAs[A](key: String)(implicit codec: TextCodec[A]): Option[A] = get(key).flatMap(codec.decode)
```

Comment:
I think there could be needs to distinguish between the absence of the parameter and decoding failure. What do you think?

Reply:
It's good point, but I'm afraid that if we have output type like `Either[String, Option[A]]` or `Option[Either[String, A]]`, it may be even harder to use than separating it to `get` and `decode` on the call site. Maybe I should define separate typed error for this, something like `DecodingError` and `NotFoundError` and then result would be `Either[QueryParamError, A]`?

Reply:
I agree with your concern. Personally I'd prefer the latter.